WorkoutGeneratorApp/App.tsx

```diff
@@ -7,56 +7,35 @@
  *
  * @format
  */
-
-import React from 'react';
-import {
-  Text,
-  View,
-} from 'react-native';
-
-import { Provider as PaperProvider } from 'react-native-paper';
-
-import { NavigationContainer } from '@react-navigation/native';
-import { createNativeStackNavigator  } from '@react-navigation/native-stack';
-import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+// import { NavigationContainer } from '@react-navigation/native';
+import * as React from 'react';
+import { BottomNavigation } from 'react-native-paper';
 
 import WorkoutGeneratorScreen from './screens/WorkoutGeneratorScreen/WorkoutGeneratorScreen';
 import SavedWorkoutsScreen from './screens/SaveScreens/SavedWorkoutsScreen';
 import SavedFiltersScreen from './screens/SaveScreens/SavedFiltersScreen';
 
+const App = () => {
+  const [index, setIndex] = React.useState(0);
+  const [routes] = React.useState([
+    { key: 'workoutgenerator', title: 'Workout Generator', icon: 'home' },
+    { key: 'savedworkouts', title: 'Saved Workouts', icon: 'floppy' },
+    { key: 'savedfilters', title: 'Saved Filters', icon: 'filter' },
+  ]);
 
+  const renderScene = BottomNavigation.SceneMap({
+    workoutgenerator: WorkoutGeneratorScreen,
+    savedworkouts: SavedWorkoutsScreen,
+    savedfilters: SavedFiltersScreen,
+  });
 
-const Tab = createBottomTabNavigator();
-
-export default function App() {
   return (
-    <PaperProvider>
-      <NavigationContainer>
-        <Tab.Navigator>
-          <Tab.Screen name="Workout Generator" component={WorkoutGeneratorScreen}/>
-          <Tab.Screen name="Saved Workouts" component={SavedWorkoutsScreen}/>
-          <Tab.Screen name="Quick Filters" component={SavedFiltersScreen}/>
-        </Tab.Navigator>
-      </NavigationContainer>
-    </PaperProvider>
+    <BottomNavigation
+    navigationState={{ index, routes }}
+    onIndexChange={setIndex}
+    renderScene={renderScene}
+  />
```

Comment:
Although I like the more explicit navigator, I can roll with this. Like the dynamic usage of useState().

Reply:
The explicit navigator was difficult to customize. It required a ton of extra code just to get icons, which also required extra libraries. Using this saved time and effort (i.e. importing more packages, etc.)